src/executor/Executor.cc

```diff
@@ -68,6 +70,11 @@ void Executor::executeChild() {
         listener->onPostForkChild();
     }
 
+    withErrnoCheck(
+            "chdir to " + childProgramWorkingDir_,
+            chdir,
+            childProgramWorkingDir_.data());
```

Comment:
Hmm so apparently since C++11 `data()` is the same as `c_str()` but elsewhere in the code we usually use `c_str()` not `data()` so can you use `c_str()` here too for consistency?